Modify rule S6194  Improve rule description for coroutine cognitive complexity CPP-4992 (#4533)
SonarSource/rspec: rules/S6194/rule.adoc

== Why is this an issue?

> Cognitive Complexity is a measure of how hard the control flow of a function is to understand. Coroutines with high Cognitive Complexity will be difficult to maintain.
> Code with high cognitive complexity is hard to read, understand, test, and modify.

Reviewer: ```suggestion
"Cognitive complexity" is a measure of how hard the control flow of a function is to understand. Code with high cognitive complexity is hard to read, understand, test, and modify.

This rule raises on coroutines with high cognitive complexity.
```
